### ISSUE TITLE
Setup CI/CD

### ISSUE THREAD

**Issue by mahesh1996** [mahesh1996/spring-security-rest#3]

**Comment by kshitij1708** [comment]
@mahesh1996 Can you assign this to me?

**Comment by mahesh1996** [comment]
@kshitij1708 Yes, you can pick it. The idea is to setup CI/CD to automate maven publishing and docs publishing to GitHub pages. I am going to use Ascii-doctor (You can say the advanced version of Markdown) for writing docs. There is [maven-asciidoctor](https://github.com/asciidoctor/asciidoctor-maven-plugin) plugin available  which converts `.adoc` files into html. So overall idea is to setup CI/CD which will build docs from `.adoc` and push it to Github pages version wise. Docs for dev can be pushed to `latest` directory. You can refer [this](https://github.com/alvarosanchez/grails-spring-security-rest/) project which uses similar kind of CI/CD using gradle. We can integrate unit test into CI/CD when it's available. You can start on this if you want.

**Comment by mahesh1996** [comment]
This will benefit auto-publishing of jar and docs whenever we release the new version or add a feature into `dev` branch.
